frontend/admin/src/js/components/apiManagedTable/basicTableHelpers.tsx

```diff
@@ -11,11 +19,93 @@ export type ColumnsOf<T extends Record<string, unknown>> = Array<Column<T>>;
 // Information about a column in a table
 export type Column<T> = {
   label: string;
+  header?: React.ReactNode;
   id: IdType<T>;
   accessor: (object: T) => React.ReactNode;
   sortColumnName?: string; // API requires exact DB column name for sorting, not the name of the field in the API 😢
 };
 
+// This defines the row selection column
+export function rowSelectionColumn<T extends RecordWithId>(
+  intl: IntlShape,
+  selectedRows: T[],
+  pageSize: number,
+  labelAccessor: (item: T) => string,
+  onRowSelectionChange: (e: RowSelectedEvent<T>) => void,
+): Column<T> {
+  return {
+    label: intl.formatMessage({
+      defaultMessage: "Row Selection",
+      description:
+        "Label for the row-selection column in the tables column-selection modal.",
+    }),
+    header: (
+      <CheckButton
+        color="white"
+        checked={selectedRows.length === pageSize}
+        indeterminate={
+          selectedRows.length > 0 && selectedRows.length < pageSize
+        }
+        label={intl.formatMessage({
+          defaultMessage: "Select/Unselect all",
```

Comment:
Should this be 'Select/Deselect all' to match the language used in the CheckButton component?

Reply:
Fixed with [update select header text](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3279/commits/b6605fc79bcc67da0a573a36fe9b989149d815ae)